cores/esp8266/core_esp8266_postmortem.cpp

```diff
@@ -128,6 +151,67 @@ void __wrap_system_restart_local() {
+            struct EXC_CONTEXT *exc_context;
+            asm volatile("rsr.excsave2 %0\n\t" : "=a" (exc_context) :: "memory");
+            // Use Special Register values from before the `ill` instruction.
+            // Note, for a double exception the exccause is that of the second exception.
+            // epc1 is still the address of the 1st exception, its cause is now unknown.
+            rst_info.exccause = exc_context->exccause;
+            rst_info.epc1 = exc_context->epc1;
+            // Double Exception
+            if (rst_info.depc) {
+                ets_printf_P(PSTR("\nDouble Exception"));
+                // The current "ESP Exception Decoder" does not process the value in depc
+                // Set a fake epc1 so the decoder identifies the line of the Double Exception
+                rst_info.epc1 = rst_info.depc;
+                // The "ESP Exception Decoder" is also helped by updating the
+                // context saved on the stack.
+                exc_context->epc1 = rst_info.depc;
+            }
+            // BP - Debug Exception
+            else if (rst_info.epc2) {
+                if (rst_info.epc2 == 0x4000dc4bu) {
+                    // Unhandled Exceptions land here. 'break 1, 1' in _xtos_unhandled_exception
+                    if (20u == exc_context->exccause) {
+                        // This could be the result from a jump or call; a call
+                        // is more likely. eg. calling a null callback function
+                        // For the call case, a0 is likely the return address.
+                        rst_info.epc1 = exc_context->excsave1;
+                    }
+                    ets_printf_P(PSTR("\nXTOS Unhandled exception - BP"));
+                }
+                else if (rst_info.epc2 == 0x4000dc3cu) {
+                    // Unhandled interrupts land here. 'break 1, 15' in _xtos_unhandled_interrupt
+                    ets_printf_P(PSTR("\nXTOS Unhandled interrupt - BP"));
+                    // Don't know what should be done here if anything.
+                }
+                else {
+                    ets_printf_P(PSTR("\nDebug Exception"));
+                    // The current Exception Decoder does not process the value in epc2
+                    // Set a fake epc1 so the Exception Decoder identifies the correct BP line
+                    rst_info.epc1 = rst_info.epc2;
+                }
+            }
+            // Kernel Exception
+            else {
+                ets_printf_P(PSTR("\nKernel Exception"));
+            }
+            ets_printf_P(PSTR(" - excsave1=0x%08x excsave2=0x%08x epc1=0x%08x\n"),
+                exc_context->excsave1, exc_context->excsave2, exc_context->epc1 );
+        }
+#endif
+        // The GCC divide routine in ROM jumps to the address below and executes ILL (00 00 00) on div-by-zero
+        // In that case, print the exception as (6) which is IntegerDivZero
+        if (rst_info.epc1 == 0x4000dce5u) {
```

Comment:
Per the comment above, this magic number seema to be a ROM address. What is there? a 0x0? Can the value be moved into a constexpr and commented?

Reply:
Yes, that address has the `ill` (illegal) instruction. Added comments and `constexpr`.